app/controllers/admin/attachments_controller.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module Admin
+  class AttachmentsController < ApplicationController
+    def index
+      @attachments = Attachment.all
+    end
+
+    def show
+      @attachment = Attachment.find(params[:id])
+    end
+
+    def new
+      @attachment = Attachment.new
+    end
+
+    def create
+      @attachment = Attachment.new(attachment_params)
+      if @attachment.save
+        redirect_to admin_attachments_path
+      else
+        render :new
+      end
+    end
+
+    def edit
+      @attachment = Attachment.find(params[:id])
+    end
+
+    def update
+      @attachment = Attachment.find(params[:id])
```

Comment:
should we use instance variable here?

Reply:
of course yes, what problems you have with instance variables? Code is fine. In case of error form must be rendered with same object